hatch_hooks.py

```diff
@@ -0,0 +1,30 @@
+import os
+
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+from hatchling.plugin import hookimpl
+
+
+class Psycopg2NoBinary(BuildHookInterface):
+    """
+    Custom build hook to install psycopg2 instead of psycopg2-binary based on the presence of `DBT_PSYCOPG2_NAME`.
```

Comment:
I was under the impression that this Hatch hook hadn't been working as expected - Is that still the case? Is it different for source vs. build/wheel distributions?

Reply:
Either way, could you add a note to this repository's README about how a user might install `psycopg2` instead of `psycopg2-binary`?

Reply:
I'll add the note.

I was playing around with different approaches using build hooks and the post-install-command config in `hatch` to try and mimic the previous behavior. I couldn't find anything that worked though. I'll get this working with the manual workaround for `psycopg2` for now and leave the automation of the manual workaround as a stretch.